### DISCUSSION TITLE
What is the usage of problem input?

### DISCUSSION THREAD

**Discussion by Jack-ReframeSystems** [tesseract-robotics/tesseract_planning#435] [Q&A] [answered]
I see that all the examples are setting the input to the task composer problem with their composite instruction (i.e. the move they want to plan):
```
TaskComposerProblem problem;
problem->input = my_composite_instruction;
```
But, in the code for the task nodes (example [MinLengthTask](https://github.com/tesseract-robotics/tesseract_planning/blob/master/tesseract_task_composer/planning/src/nodes/min_length_task.cpp#L82)) the instruction is read from the `context->data_storage->input` field. This confused me when working with Tesseract as the data storage argument to `TaskComposerExecutor::run` is optional

I therefore assumed that the `problem->input` field, which all the examples used, would be the correct input. But, this didn't turn out to be case. 

What then is the purpose of the `problem->input` field?

**Comment by Levi-Armstrong** [comment, marked as answer]
The input should be the input to the task composer pipeline and the first task in the pipeline should process this input and populate the data storage and all other tasks should interface with the data storage. This provides a clean separation between the input data to the pipeline and the data generated within the pipeline. The plan is to remove the data storage object as an input when running a pipeline to avoid the confusion that you pointed out.
